src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -5605,7 +5605,7 @@ let CheckOneImplFile
                 implFileTypePriorToSig |> IterTyconsOfModuleOrNamespaceType (fun tycon ->
                     FinalTypeDefinitionChecksAtEndOfInferenceScope (cenv.infoReader, envAtEnd.NameEnv, cenv.tcSink, true, denvAtEnd, tycon))
 
-            with exn ->
+            with exn when not exn.IsOperationCancelled ->
```

Comment:
The situation with error recovery is spread all over the place, what about encapsulation the logic for recovery-relevant or irrelevant? e.g. in an struct active pattern:

```fsharp

[<return: Struct>]
let (|RecoveryRelevantExn|_|) (exn:Exception) =
    if exn.IsOperationCancelled then
        ValueNone
    else ValueSome exn

let tryStuff() =
    try
        5 / 0
    with RecoveryRelevantExn e -> 42

```

Reply:
Fixed, thanks.